src/app/devices/deviceContent/sagas/digitalTwinSaga.ts

```diff
@@ -16,10 +20,63 @@ export function* getDigitalTwinSaga(action: Action<string>) {
             digitalTwinId: action.payload,
         };
 
-        const digitalTwinInterfaceProperties = yield call(fetchDigitalTwin, parameters);
+        const digitalTwin = yield call(fetchDigitalTwin, parameters);
 
-        yield put(getDigitalTwinAction.done({params: action.payload, result: digitalTwinInterfaceProperties}));
+        yield put(getDigitalTwinAction.done({params: action.payload, result: digitalTwin}));
     } catch (error) {
         yield put(getDigitalTwinAction.failed({params: action.payload, error}));
     }
 }
+
+export function* patchDigitalTwinSaga(action: Action<PatchDigitalTwinActionParameters>) {
+    try {
+        const connectionString = yield call(getActiveAzureResourceConnectionStringSaga);
+        const parameters: PatchDigitalTwinParameters = {
+            ...action.payload,
+            connectionString
+        };
+
+        const responseCode = yield call(patchDigitalTwinAndGetResponseCode, parameters);
+        const digitalTwin = yield call(fetchDigitalTwin, {connectionString, digitalTwinId: parameters.digitalTwinId});
+
+        if (responseCode === DataPlaneStatusCode.Accepted) {
+            yield put(addNotificationAction.started({
+                text: {
+                    translationKey: ResourceKeys.notifications.patchDigitalTwinOnAccept,
+                    translationOptions: {
+                        deviceId: action.payload.digitalTwinId
+                    },
+                },
+                type: NotificationType.success
+            }));
+            yield put(patchDigitalTwinAction.done({params: action.payload, result: digitalTwin}));
+        }
+        else if (responseCode === DataPlaneStatusCode.SuccessLowerBound) {
+            yield put(addNotificationAction.started({
+                text: {
+                    translationKey: ResourceKeys.notifications.patchDigitalTwinOnSuccess,
```

Comment:
Only differs from the above by the value of translationKey, should be able to combine these two blocks of code.

Reply:
Yeah totally.